src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer.Common/DefaultRemoteTextLoaderFactory.cs

```diff
@@ -78,6 +78,12 @@ public override Task<TextAndVersion> LoadTextAndVersionAsync(Workspace workspace
                     // in us trying to refresh the "closed" files buffer with what's on disk; however, there's nothing actually on disk because the file was renamed.
                     textAndVersion = TextAndVersion.Create(SourceText.From(string.Empty), VersionStamp.Default, filePath: _filePath);
                 }
+                catch (DirectoryNotFoundException)
+                {
+                    var directory = Path.GetDirectoryName(_filePath);
+                    Directory.CreateDirectory(directory);
```

Comment:
Makes sense, I've updated to just use the existing FileNotFound logic. I'm assuming this is what you meant by:
> just catch the dir not found exceptions and let them happen repeatedly

Let me know if otherwise. Reluctant to return null as it was complicating things higher in the event stack.